src/main/java/org/folio/edge/core/utils/OkapiClientFactoryInitializer.java

```diff
@@ -0,0 +1,55 @@
+package org.folio.edge.core.utils;
+
+import static org.folio.edge.core.Constants.SYS_OKAPI_URL;
+import static org.folio.edge.core.Constants.SYS_REQUEST_TIMEOUT_MS;
+import static org.folio.edge.core.Constants.SYS_WEB_CLIENT_KEY_ALIAS;
+import static org.folio.edge.core.Constants.SYS_WEB_CLIENT_KEY_ALIAS_PASSWORD;
+import static org.folio.edge.core.Constants.SYS_WEB_CLIENT_SSL_ENABLED;
+import static org.folio.edge.core.Constants.SYS_WEB_CLIENT_TRUSTSTORE_PASSWORD;
+import static org.folio.edge.core.Constants.SYS_WEB_CLIENT_TRUSTSTORE_PATH;
+import static org.folio.edge.core.Constants.SYS_WEB_CLIENT_TRUSTSTORE_PROVIDER;
+import static org.folio.edge.core.Constants.SYS_WEB_CLIENT_TRUSTSTORE_TYPE;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.net.KeyStoreOptions;
+import io.vertx.core.net.TrustOptions;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class OkapiClientFactoryInitializer {
+  private static final Logger logger = LogManager.getLogger(OkapiClientFactoryInitializer.class);
+
+  private OkapiClientFactoryInitializer() {
+  }
+
+  public static OkapiClientFactory createInstance(Vertx vertx, JsonObject config) {
+    String okapiUrl = config.getString(SYS_OKAPI_URL);
+    Integer requestTimeout = config.getInteger(SYS_REQUEST_TIMEOUT_MS);
+    boolean isSslEnabled = config.getBoolean(SYS_WEB_CLIENT_SSL_ENABLED);
+    if (isSslEnabled) {
+      logger.info("Creating OkapiClientFactory with Enhance HTTP Endpoint Security and TLS mode enabled");
+      String truststoreType = config.getString(SYS_WEB_CLIENT_TRUSTSTORE_TYPE);
+      String truststoreProvider = config.getString(SYS_WEB_CLIENT_TRUSTSTORE_PROVIDER);
+      String truststorePath = config.getString(SYS_WEB_CLIENT_TRUSTSTORE_PATH);
+      String truststorePassword = config.getString(SYS_WEB_CLIENT_TRUSTSTORE_PASSWORD);
+      String keyAlias = config.getString(SYS_WEB_CLIENT_KEY_ALIAS);
+      String keyAliasPassword = config.getString(SYS_WEB_CLIENT_KEY_ALIAS_PASSWORD);
+      if (truststoreType != null && truststorePath != null && truststorePassword != null) {
```

Comment:
maybe we should use StringUtils in case of user send parameter as " " (empty string)

Reply:
I thought about it, there is no StringUtils in classpath from commons.lang3, so all other string params are checking with just non null. So remained consistent with other params

Reply:
Can be swapped to Objects.nonNull()

Reply:
In this case I will suggest to add apache common lang dependency to get StringUtils methods 
or manually check. Checking after null avoid NullpointerException
```
    if (truststoreType != null && !truststoreType.isBlank() 
        && truststorePath != null && !truststorePath.isBlank()
        && truststorePassword != null && !truststorePassword.isBlank()) {
```

Reply:
used utility method from aws sdk